pkg/util/trivy/trivy.go

```diff
@@ -337,7 +337,7 @@ func (c *Collector) scan(ctx context.Context, artifact artifact.Artifact, applie
 		if err != nil {
 			return nil, err
 		}
-		c.cache.setKeysForEntity(imgMeta.EntityID.ID, append(artifactReference.BlobIDs, artifactReference.ID))
+		cache.setKeysForEntity(imgMeta.EntityID.ID, append(artifactReference.BlobIDs, artifactReference.ID))
```

Comment:
What is the difference ?

Reply:
I think the issue is with the check on 332. You need to use the same variable in both lines

Reply:
`cache` is an in memory cache that is thrown away after the scan operation. When scanning from file system, the persistent cache is not initialized because it is not useful. 
https://github.com/DataDog/datadog-agent/blob/f57fb570f5660d35b5243f6bc067f25615a26682/pkg/util/trivy/trivy.go#L306-L307

Basically, to compute the cache key we need the scan result. So a cache hit does not result in computation gain.
I think we need another layer of caching.

I thought I fixed this and renamed `c.cache` as `c.persistentCache` but didn't push it